docs/user-guide/isis/zoom.ipynb

```diff
@@ -175,6 +174,35 @@
     "        pipeline.insert(provider)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9f942232",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cyclebane as cb\n",
+    "\n",
+    "\n",
+    "def merge(*dicts: dict) -> dict:\n",
+    "    return {k: v for d in dicts for k, v in d.items()}\n",
+    "\n",
+    "\n",
+    "def make_workflow(\n",
+    "    pipeline: sciline.Pipeline, masks: list[str]\n",
+    ") -> sciline.task_graph.TaskGraph:\n",
+    "    graph = pipeline.get(\n",
+    "        IofQ[SampleRun], handler=sciline.HandleAsComputeTimeException()\n",
+    "    )\n",
+    "    cbgraph = cb.Graph(graph.to_networkx())\n",
+    "    cbgraph[MaskedDetectorIDs] = (\n",
+    "        cbgraph[MaskedDetectorIDs]\n",
+    "        .map({PixelMaskFilename: masks})\n",
+    "        .reduce(attrs={\"reduce\": merge})\n",
+    "    )\n",
+    "    return sciline.task_graph.TaskGraph.from_networkx(cbgraph.to_networkx())"
```

Comment:
Just to understand what's going on here, is this what it does?
```python
masked_detector_ids = dict()
for pixel_mask_filename in masks:
    pl[PixelMaskFilename] = pixel_mask_filename
    masked_detector_ids = merge(masked_detector_ids, pl.compute(MaskedDetectorIDs))
pl[MaskedDetectorIDs] = masked_detector_ids
```
But I guess it's not the same because in the `sciline-cyclebane` version the execution is delayed.

Reply:
Yes, I think this is a good way of putting it (if I understand your code correctly, i.e., compute the branch in each case, merge all the result into a single dict, and set it as a value).